requirements/edx/base.in

```diff
@@ -75,7 +75,7 @@ edx-api-doc-tools
 edx-bulk-grades                     # LMS REST API for managing bulk grading operations
 edx-ccx-keys
 edx-celeryutils
-edx-completion
+git+https://github.com/regisb/completion.git@regisb/deprecate-waffle-namespace#egg=edx-completion==3.2.6
```

Comment:
This PR should work against 3.2.5, right?  Let's leave completion as-is, and upgrade it later when we are going to 2.0.0 for edx-toggles.

Reply:
Well, no, because this PR depends on edx-toggles 2.0.0, which edx-completion==3.2.5 is not compatible with.

Reply:
See discussion in Slack. I don’t see why we can’t get off everything deprecated in advance of a 2.0.0 breaking-change upgrade.

Reply:
Right! Still, I think it makes sense to upgrade to 4.0.0 to benefit from the new API. This is possible because completion==4.0.0 no longer depends on edx-toggles==2.0.0. I might be able to create a separate PR for that, though. I'll give it a try.

Reply:
Here is the PR: https://github.com/edx/edx-platform/pull/25657

Reply:
Well, turns out it's not possible to mix WaffleSwitch/Flag objects with different APIs, because of the toggle state API endpoint: https://build.testeng.edx.org/job/edx-platform-python-pipeline-pr/23786/testReport/junit/openedx.core.djangoapps.waffle_utils.tests.test_views/ToggleStateViewTests/Run_Tests___cms_unit___test_code_owners_without_module_information/

I guess I could make this view forward-compatible, but it would be a real ugly fix. I'm for upgrading completion at the same time as we move to new-style waffle objects.

Reply:
To confirm my understanding, this plan still includes using `__futures__`, right?  Just at a later time?

Reply:
> To confirm my understanding, this plan still includes using __futures__, right? Just at a later time?

I'm not sure what you mean. If we upgrade completion to v4.0.0 now (which I suggest we do) there will be no need to make the toggle state view capable of handling both old and new Waffle objects.

Reply:
I’m not sure what I meant. I just want to make all changes are made under toggles 1.2.0 first, and I think we are now on that path. Marking as resolved.